javascript/ql/src/semmle/javascript/security/dataflow/RemoteFlowSources.qll

```diff
@@ -10,8 +10,19 @@ import semmle.javascript.security.dataflow.DOM
 abstract class RemoteFlowSource extends DataFlow::Node {
   /** Gets a string that describes the type of this remote flow source. */
   abstract string getSourceType();
-}
 
+  /**
+   * Holds if this flow source comes from an incoming request, and this part of the
+   * request can be controlled by a third party, that is, an actor other than the one
+   * sending the request.
+   *
+   * Any web site can redirect the visitor's browser to any other domain, and in doing so control
```

Comment:
I don't understand this explanation, particularly the bit beginning with "and in doing so". Why is it important what the web site controls? Aren't we more interested in what data the remote user/attacker controls?

Reply:
The web site is the attacker here. Would this clarify things?

> A malicious web site can redirect the visitor's browser to any other domain [...]

Reply:
The goal of the paragraph is to explain why a user might sent a request with data he doesn't control, and why it's not okay to just blame for user for sending stupid requests that compromise his account.

Reply:
Yes, I think that would be clearer.